File: src/main/kotlin/com/ymougenel/bgaranking/models/dto/RankVariation.java
package com.ymougenel.bgaranking.models.dto;

import com.ymougenel.bgaranking.models.Ranking;

public class RankVariation {
    public long elo;
    public long rank;
    public long rankVariation;
    public long eloVariation;

    public RankVariation(Ranking before, Ranking after) {
        this.elo = 0;
        this.rank = after.getRank();
        this.eloVariation = 0;
        this.rankVariation = after.getRank() - before.getRank();
    }
}

File: src/main/kotlin/com/ymougenel/bgaranking/models/dto/RankVariation.kt
package com.ymougenel.bgaranking.models.dto

import com.ymougenel.bgaranking.models.Ranking

class RankVariation(before: Ranking, after: Ranking) {
    var elo: Long = 0
    var rank: Long = 0
    var rankVariation: Long = 0
    var eloVariation: Long = 0

    init {
        this.elo = 0
        this.rank = after.rank
        this.eloVariation = 0
        this.rankVariation = after.rank - before.rank
    }
}
